browser/registry_test.go

```diff
@@ -296,6 +293,48 @@ func TestBrowserRegistry(t *testing.T) {
 
 		assert.True(t, browserRegistry.stopped.Load())
 	})
+
+	// This test ensures that the chromium browser's lifecycle is not controlled
+	// by the vu context.
+	t.Run("dont_close_browser_on_vu_context_close", func(t *testing.T) {
+		t.Parallel()
+
+		vu := k6test.NewVU(t)
+		var cancel context.CancelFunc
+		vu.CtxField, cancel = context.WithCancel(vu.CtxField)
+		browserRegistry := newBrowserRegistry(vu, remoteRegistry, &pidRegistry{}, nil)
+
+		vu.ActivateVU()
+
+		// Send a few IterStart events
+		vu.StartIteration(t, k6test.WithIteration(0))
+
+		// Verify browsers are initialized
+		browserRegistry.mu.RLock()
+		assert.Equal(t, 1, len(browserRegistry.m))
+		browserRegistry.mu.RUnlock()
+
```

Comment:
I'd move this to a method in `browserRegistry` as the test is coupled to the registry's internals and becomes unnecessarily lengthy because of mutex management (which is supposed to be the registry's job).

Reply:
So create a method like `getBrowserCount`? It would only be used by the test. I've added it in f3719419bc805b165c9d488d43ebec8c5eb2ebaf, and also done it for iteration traces too in ac2df75082aab2e644be878e8c4131f5787bea9f.

Reply:
The only user for the test is fine. I'd rename `getIterationTracesCount` to `iterationTracesCount` as "get" is frowned upon in idiomatic Go (or leave it as it is, no strong opinion :).

Reply:
No problem, changed in 89bcc73bf88660cdd1d1de62de76bcf1d093046c :)